iphone/Classes/GeolocationModule.m

```diff
@@ -869,7 +869,9 @@ -(void)requestLocationPermissions:(id)args
                (currentPermissionLevel == kCLAuthorizationStatusAuthorized)) {
                 errorMessage = @"Cannot change already granted permission from AUTHORIZATION_ALWAYS to AUTHORIZATION_WHEN_IN_USE";
             } else if (!userDidApproveOrReject) {
-                [[self locationPermissionManager] requestWhenInUseAuthorization];
+                dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Should use                 TiThreadPerformOnMainThread(^{
                    [[self locationPermissionManager] requestWhenInUseAuthorization];
                }, NO); instead. As it does a couple of checks regarding the main thread to prevent  crashes.